### PR TITLE
Update misc tools

### PR DESCRIPTION
**Pull request by jacob-delgado**
Holding until more tools are updated

### CODE REVIEW

**Review by ericvn — APPROVED** [review]

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ ARG ISTIO_TOOLS_SHA
 # Pinned versions of stuff we pull in, keep this list sorted alphabetically
 ENV BENCHSTAT_VERSION=9c9101da8316
 ENV BOM_VERSION=v0.6.0
-ENV BUF_VERSION=v1.30.1
+ENV BUF_VERSION=v1.31.0
 ENV COSIGN_VERSION=v2.2.4
 ENV CRANE_VERSION=v0.19.1
 ENV GCLOUD_VERSION=467.0.0
```
Comment (jacob-delgado) [inline docker/build-tools/Dockerfile:43]:
```suggestion
ENV GCLOUD_VERSION=473.0.0
```

**Comment by ericvn** [comment]
Update GH_VERSION=2.49.0

**Comment by ericvn** [comment]
And maybe DOCKER_VERSION=5:26.1.0-1~ubuntu.22.04~jammy

**Comment by jacob-delgado** [comment]
docker to `26.1.1-1~ubuntu.22.04~jammy`

buildkit to v0.13.2

gcloud to 474

**Comment by jacob-delgado** [comment]
hugo to 0.125.4

**Pull request closed without merging**